nano_press/templates/components/core/domain.html

```
@@ -0,0 +1,155 @@
+                    await verify_domain(domain, expectedIp);
+                    
+                    // Mark as verified in store
+                    this.$store.installer.domainVerified = true;
+                    this.domainStatusMessage = `✅ ${domain} resolves to ${expectedIp}`;
+                    
+                    console.log('✅ Domain verification successful');
+                    
+                    setTimeout(() => {
+                        this.$store.installer.currentStep = 4;
+                    }, 1000);
+                } catch (err) {
+                    console.error('❌ Domain verification failed:', err);
+                    this.$store.installer.domainVerified = false;
+                    this.domainStatusMessage = err.message || "Error verifying domain.";
+                } finally {
+                    this.isLoading = false;
+                }
+            } else {
+                // No domain, just continue
+                this.$store.installer.domain = '';
+                this.$store.installer.domainVerified = false;
+                this.$store.installer.currentStep = 4;
+            }
+        }
+    };
+}
+
+async function verify_domain(domain, expectedIp) {
+    const url = '/api/method/nano_press.api.check_domain_resolves_to_ip';
+    try {
+        const payload = {
+            domain: domain,
+            expected_ip: expectedIp
+        };
+
+        const response = await frappe_call(url, payload, "POST");
+
+        if (!response?.message?.success) {
+            throw new Error(response?.message?.message || "Domain verification failed");
+        }
+
+        const message = response.message.message;
+        toast(message, { type: "success" });
+        return response.message;
+    } catch (err) {
+        const message = err.message || "Failed to verify domain";
+        toast(message, { type: "danger" });
+        throw err;
+    }
+}
```

See if you went through previous comments I mentioned why to use frappe.throw instead of return statement, then you don't have to manually throw error in line 143 even in the backend code for
`check_domain_resolves_to_ip`
You can use frappe.throw instead of return { "success" : False} then it will be much cleaner.

Please create a separate issue for it will tackle this later